incubator/hnc/Makefile

```diff
@@ -247,7 +247,35 @@ kind-deploy:
 # Go test suite has a 10m timeout, and the flag disables that timeout (as of
 # July 2020, these tests take ~15m and that number is expected to grow).
 .PHONY: test-e2e
-test-e2e:
+test-e2e: before-e2e
+	go clean -testcache
+	go test -v -timeout 0 ./test/e2e/...
+
+# This batch test will run e2e tests N times on the current cluster the user
+# deployed (either kind or a kubernetes cluster), e.g. "make test-e2e-batch N=10"
+.PHONY: test-e2e-batch
+test-e2e-batch: before-e2e
+	number=1 ; while [[ $$number -le $N ]] ; do \
+		echo $$number ; \
+    ((number = number + 1)) ; \
+		go clean -testcache ; \
+		go test -v -timeout 0 ./test/e2e/... ; \
+	done
+
+before-e2e:
+	@echo
+	@echo "Ensuring excluded-namespaces are properly configured..."
+	@echo
+	kubectl label ns hnc-system hnc.x-k8s.io/excluded-namespace=true --overwrite
+	@echo
+	kubectl label ns kube-node-lease hnc.x-k8s.io/excluded-namespace=true --overwrite
+	@echo
+	kubectl label ns kube-public hnc.x-k8s.io/excluded-namespace=true --overwrite
+	@echo
+	kubectl label ns kube-system hnc.x-k8s.io/excluded-namespace=true --overwrite
+	@echo
+	@echo "If you see the above excluded namespaces are just labeled, it means your HNC installation was not complete."
```

Comment:
What does this mean? The first time you run this, you _should_ expect to see the namespaces being labeled, right?

Reply:
Regular users shouldn't be using the makefile so I wouldn't worry about "installing" HNC, so you can assume that this will only be read by developers who are willing to open up the makefile. E.g. you could do something like:

```
@echo Ensuring all system namespaces are excluded from HNC
@kubectl label ns ...
...
```